histomicsui/web_client/panels/DrawWidget.js

```diff
@@ -756,6 +766,38 @@ var DrawWidget = Panel.extend({
         }
     },
 
+    /**
+     * Show or hide width/height input depending on the currently selected drawing mode.
+     */
+    _updateConstraintValueInputs() {
+        if (['fixed_aspect_ratio', 'fixed_size'].includes(this.$('.h-fixed-shape:checked').attr('mode'))) {
+            this.$('.h-fixed-values').show();
+        } else {
+            this.$('.h-fixed-values').hide();
+        }
+    },
+
+    /**
+     * Update the width/height constraint for a shape being drawn with a fixed
+     * aspect ratio or fixed size.
+     */
+    _changeShapeConstraint(evt) {
+        const opts = {
+            size_mode: this.$('.h-fixed-shape:checked').attr('mode'),
+            fixed_width: parseFloat(this.$('.h-fixed-width').val()),
+            fixed_height: parseFloat(this.$('.h-fixed-height').val())
+        };
+        this._saveEditOptions(opts);
+
+        this._updateConstraintValueInputs();
+
+        if (opts.size_mode === 'fixed_aspect_ratio') {
+            this.viewer.startDrawMode(this._drawingType, {constraint: opts.fixed_width / opts.fixed_height});
+        } else if (opts.size_mode === 'fixed_size') {
+            this.viewer.startDrawMode(this._drawingType, { constraint: { width: opts.fixed_width, height: opts.fixed_height } });
```

Comment:
@manthey this isn't finished, I still have some minor styling issues to iron out and also have to make this work with ellipses as well (it currently only works with rectangles). But, I'm running into the same problem I mentioned [here](https://github.com/DigitalSlideArchive/HistomicsUI/issues/26#issuecomment-1144087622).
I tried specifying a constraint like you said [here](https://github.com/DigitalSlideArchive/HistomicsUI/issues/26#issuecomment-1126089848), but it doesn't seem to work. The drawing tool behaves the same as it normally does for me, no matter what the `constraint` option is set to. Any ideas?